### PR TITLE
[BIP-303] zkEVM Permissioning BIP

### PR DESCRIPTION
**Pull request by solarcurvey**
goal of this BIP is to not only grant all necessary permissions to bring zkEVM online but to do all the setup tasks as well. 

once we get the json for granting the base permissions we can load all the json's in the safe to sim on tenderly

### CODE REVIEW

**Comment by solarcurvey** [comment]
can refer to this checklist to understand what 303A, 303B, 303C should be doing.

note, tenderly does not support zkEVM. so we're flyin blind

File: BIPs/BIP-303/zkevm.json [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    {
+      "to": "0xA331D84eC860Bf466b4CdCcFb4aC09a1B43F3aE6",
+      "value": "0",
+      "data": null,
+      "contractMethod": {
+        "inputs": [
+          { "internalType": "bytes32[]", "name": "roles", "type": "bytes32[]" },
+          { "internalType": "address", "name": "account", "type": "address" }
+        ],
+        "name": "grantRoles",
```
Comment (Tritium-VLK) [inline BIPs/BIP-303/zkevm.json:75]:
```
>>> roleList = "0x1282ab709b2b70070f829c46bc36f76b32ad4989fecb2fcb09a1b3ce00bbfc30, 0xc149e88b59429ded7f601ab52ecd62331cac006ae07c16543439ed138dcb8d34, 
0x78ad1b68d148c070372f8643c4648efbb63c6a8a338f3c24714868e791367653, 0xeba777d811cd36c06d540d7ff2ed18ed042fd67bbf7c9afcf88c818c7ee6b498, 0x0014a06d322ff0
7fcc02b12f93eb77bb76e28cdee4fc0670b9dec98d24bbfec8, 0x7b8a1d293670124924a0f532213753b89db10bde737249d4540e9a03657d1aff"
roleList = roleList.split(", ")
tx = Authorizer.grantRoles(roleList, BatchRelayer.address, {'from': daomsig})
tx.call_trace(expand=True)
 
Transaction sent: 0x1d05b15ca5b2a53fb1662d5c19584a5e982d48f2e3716484d20f61cc996ebb55
  Gas price: 0.0 gwei   Gas limit: 20000000   Nonce: 4
  Authorizer.grantRoles confirmed   Block: 387615   Gas used: 431853 (2.16%)

Call trace for '0x1d05b15ca5b2a53fb1662d5c19584a5e982d48f2e3716484d20f61cc996ebb55':
Initial call cost  [24748 gas]
Authorizer.grantRoles  0:1706  [1367 / 407105 gas]
├── AccessControl.grantRole  194:416  [953 / 67623 gas]
│   ├── AccessControl.hasRole  216:264  [118 / 1023 gas]
│   │   └── EnumerableSet.contains  233:255  [905 gas]
│   ├── _require  268:276  [30 gas]
│   └── AccessControl._grantRole  282:412  [2038 / 65617 gas]
│       └── EnumerableSet.add  299:390  [62674 / 63579 gas]
│           └── EnumerableSet.contains  306:328  [905 gas]
├── AccessControl.grantRole  448:670  [953 / 67623 gas]
│   ├── AccessControl.hasRole  470:518  [118 / 1023 gas]
│   │   └── EnumerableSet.contains  487:509  [905 gas]
│   ├── _require  522:530  [30 gas]
│   └── AccessControl._grantRole  536:666  [2038 / 65617 gas]
│       └── EnumerableSet.add  553:644  [62674 / 63579 gas]
│           └── EnumerableSet.contains  560:582  [905 gas]
├── AccessControl.grantRole  702:924  [953 / 67623 gas]
│   ├── AccessControl.hasRole  724:772  [118 / 1023 gas]
│   │   └── EnumerableSet.contains  741:763  [905 gas]
│   ├── _require  776:784  [30 gas]
│   └── AccessControl._grantRole  790:920  [2038 / 65617 gas]
│       └── EnumerableSet.add  807:898  [62674 / 63579 gas]
│           └── EnumerableSet.contains  814:836  [905 gas]
├── AccessControl.grantRole  956:1178  [953 / 67623 gas]
│   ├── AccessControl.hasRole  978:1026  [118 / 1023 gas]
│   │   └── EnumerableSet.contains  995:1017  [905 gas]
│   ├── _require  1030:1038  [30 gas]
│   └── AccessControl._grantRole  1044:1174  [2038 / 65617 gas]
│       └── EnumerableSet.add  1061:1152  [62674 / 63579 gas]
│           └── EnumerableSet.contains  1068:1090  [905 gas]
├── AccessControl.grantRole  1210:1432  [953 / 67623 gas]
│   ├── AccessControl.hasRole  1232:1280  [118 / 1023 gas]
│   │   └── EnumerableSet.contains  1249:1271  [905 gas]
│   ├── _require  1284:1292  [30 gas]
│   └── AccessControl._grantRole  1298:1428  [2038 / 65617 gas]
│       └── EnumerableSet.add  1315:1406  [62674 / 63579 gas]
│           └── EnumerableSet.contains  1322:1344  [905 gas]
└── AccessControl.grantRole  1464:1686  [953 / 67623 gas]
    ├── AccessControl.hasRole  1486:1534  [118 / 1023 gas]
    │   └── EnumerableSet.contains  1503:1525  [905 gas]
    ├── _require  1538:1546  [30 gas]
    └── AccessControl._grantRole  1552:1682  [2038 / 65617 gas]
        └── EnumerableSet.add  1569:1660  [62674 / 63579 gas]
            └── EnumerableSet.contains  1576:1598  [905 gas]
>>>

```

File: BIPs/BIP-303/zkevm.json [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        "payable": false
+      },
+      "contractInputsValues": {
+        "roles": "[0xbe2a180d5cc5d803a8eec4cea569989fc1c593d7eeadd1f262f360a68b0e842e, 0xb28b769768735d011b267f781c3be90bce51d5059ba015bc7a28b3e882fb2083]",
+        "account": "0x1802953277FD955f9a254B80Aa0582f193cF1d77"
```
Comment (Tritium-VLK) [inline BIPs/BIP-303/zkevm.json:97]:
```
>>> a.reversebook["0x1802953277FD955f9a254B80Aa0582f193cF1d77"]
'20220725-protocol-fee-percentages-provider/ProtocolFeePercentagesProvider'
>>> ProtocolFeesPercentagesProvider = Contract.from_explorer("0x1802953277FD955f9a254B80Aa0582f193cF1d77")
Fetching source of 0x1802953277FD955f9a254B80Aa0582f193cF1d77 from api-zkevm.polygonscan.com...
```

File: BIPs/BIP-303/zkevm.json [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    {
+      "to": "0xA331D84eC860Bf466b4CdCcFb4aC09a1B43F3aE6",
+      "value": "0",
+      "data": null,
+      "contractMethod": {
+        "inputs": [
+          { "internalType": "bytes32[]", "name": "roles", "type": "bytes32[]" },
+          { "internalType": "address", "name": "account", "type": "address" }
+        ],
+        "name": "grantRoles",
```
Comment (Tritium-VLK) [inline BIPs/BIP-303/zkevm.json:92]:
```
tx = Authorizer.grantRoles(roleList, ProtocolFeesPercentagesProvider.address, {'from': daomsig})
tx.call_trace(expand=True)
Transaction sent: 0xa0ec09cc98cde221e2256a2440a1ad861e98de81a3a5eafce50d0c6cc2bd61e9
  Gas price: 0.0 gwei   Gas limit: 20000000   Nonce: 6
  Authorizer.grantRoles confirmed   Block: 387617   Gas used: 128861 (0.64%)

Call trace for '0xa0ec09cc98cde221e2256a2440a1ad861e98de81a3a5eafce50d0c6cc2bd61e9':
Initial call cost  [22736 gas]
Authorizer.grantRoles  0:690  [879 / 106125 gas]
├── AccessControl.grantRole  194:416  [953 / 52623 gas]
│   ├── AccessControl.hasRole  216:264  [118 / 1023 gas]
│   │   └── EnumerableSet.contains  233:255  [905 gas]
│   ├── _require  268:276  [30 gas]
│   └── AccessControl._grantRole  282:412  [2038 / 50617 gas]
│       └── EnumerableSet.add  299:390  [47674 / 48579 gas]
│           └── EnumerableSet.contains  306:328  [905 gas]
└── AccessControl.grantRole  448:670  [953 / 52623 gas]
    ├── AccessControl.hasRole  470:518  [118 / 1023 gas]
    │   └── EnumerableSet.contains  487:509  [905 gas]
    ├── _require  522:530  [30 gas]
    └── AccessControl._grantRole  536:666  [2038 / 50617 gas]
        └── EnumerableSet.add  553:644  [47674 / 48579 gas]
            └── EnumerableSet.contains  560:582  [905 gas]

```

File: BIPs/BIP-303/zkevm.json [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+        "payable": false
+      },
+      "contractInputsValues": {
+        "roles": "[0xb2b6e48fa160a7c887d9d7a68b6a9bb9d47d4953d33e07f3a39e175d75e97796]",
+        "account": "0x230a59F4d9ADc147480f03B0D3fFfeCd56c3289a"
```
Comment (Tritium-VLK) [inline BIPs/BIP-303/zkevm.json:114]:
```
a.reversebook["0x230a59F4d9ADc147480f03B0D3fFfeCd56c3289a"]
'20220517-protocol-fee-withdrawer/ProtocolFeesWithdrawer'
>>> ProtocolFeesWithdawer=["0x230a59F4d9ADc147480f03B0D3fFfeCd56c3289a"]
```

File: BIPs/BIP-303/zkevm.json [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+    {
+      "to": "0xA331D84eC860Bf466b4CdCcFb4aC09a1B43F3aE6",
+      "value": "0",
+      "data": null,
+      "contractMethod": {
+        "inputs": [
+          { "internalType": "bytes32[]", "name": "roles", "type": "bytes32[]" },
+          { "internalType": "address", "name": "account", "type": "address" }
+        ],
+        "name": "grantRoles",
```
Comment (Tritium-VLK) [inline BIPs/BIP-303/zkevm.json:109]:
```
>>> roleList = ["0xb2b6e48fa160a7c887d9d7a68b6a9bb9d47d4953d33e07f3a39e175d75e97796"]
tx = Authorizer.grantRoles(roleList, ProtocolFeesWithdawer.address, {'from': daomsig})
tx.call_trace(expand=True)
Transaction sent: 0xc464e33da7cb0bcde96e61bf7acd30151a4961bf18f3d8bc308edd983f54ac3e
  Gas price: 0.0 gwei   Gas limit: 20000000   Nonce: 7
  Authorizer.grantRoles confirmed   Block: 387618   Gas used: 90604 (0.45%)

Call trace for '0xc464e33da7cb0bcde96e61bf7acd30151a4961bf18f3d8bc308edd983f54ac3e':
Initial call cost  [22224 gas]
Authorizer.grantRoles  0:436  [757 / 68380 gas]
└── AccessControl.grantRole  194:416  [953 / 67623 gas]
    ├── AccessControl.hasRole  216:264  [118 / 1023 gas]
    │   └── EnumerableSet.contains  233:255  [905 gas]
    ├── _require  268:276  [30 gas]
    └── AccessControl._grantRole  282:412  [2038 / 65617 gas]
        └── EnumerableSet.add  299:390  [62674 / 63579 gas]
            └── EnumerableSet.contains  306:328  [905 gas]
```

File: BIPs/BIP-303/zkevm.json [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
```
Comment (Tritium-VLK) [inline BIPs/BIP-303/zkevm.json:126]:
```
>>> roleList = ["0xbeb10dd1f094c0751dc69e30b35c3f37cee8a9303f6e7380c83d31adcba39ea8"]
tx = Authorizer.grantRoles(roleList, daomsig, {'from': daomsig})
tx.call_trace(expand=True)
Transaction sent: 0xaad3f87f1c316cab9a7505646a6cc2723d08ea34f9fce73da8c63a64d99fee0f
  Gas price: 0.0 gwei   Gas limit: 20000000   Nonce: 8
  Authorizer.grantRoles confirmed   Block: 387619   Gas used: 90604 (0.45%)

Call trace for '0xaad3f87f1c316cab9a7505646a6cc2723d08ea34f9fce73da8c63a64d99fee0f':
Initial call cost  [22224 gas]
Authorizer.grantRoles  0:436  [757 / 68380 gas]
└── AccessControl.grantRole  194:416  [953 / 67623 gas]
    ├── AccessControl.hasRole  216:264  [118 / 1023 gas]
    │   └── EnumerableSet.contains  233:255  [905 gas]
    ├── _require  268:276  [30 gas]
    └── AccessControl._grantRole  282:412  [2038 / 65617 gas]
        └── EnumerableSet.add  299:390  [62674 / 63579 gas]
            └── EnumerableSet.contains  306:328  [905 gas]
```

File: BIPs/BIP-303/BIP-303A.json [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+    {
+      "to": "0x1802953277FD955f9a254B80Aa0582f193cF1d77",
+      "value": "0",
+      "data": null,
+      "contractMethod": {
+        "inputs": [
+          { "internalType": "uint256", "name": "feeType", "type": "uint256" },
+          { "internalType": "uint256", "name": "newValue", "type": "uint256" }
+        ],
+        "name": "setFeeTypePercentage",
```
Comment (Tritium-VLK) [inline BIPs/BIP-303/BIP-303A.json:23]:
```
>>> tx=ProtocolFeesPercentagesProvider.setFeeTypePercentage(0, 500000000000000000, {'from': daomsig})
Transaction sent: 0xe7d829cadfac1d0149b81606b121696ba4876f710356807b2ca6c92857990213
  Gas price: 0.0 gwei   Gas limit: 20000000   Nonce: 9
  ProtocolFeePercentagesProvider.setFeeTypePercentage confirmed   Block: 387620   Gas used: 62079 (0.31%)

>>> tx.call_trace(expand=True)
/Users/tritium/venv/lib/python3.9/site-packages/brownie/network/contract.py:1204: BrownieCompilerWarning: 0xBA12222222228d8Ba445958a75a0704d566BF2C8: Locally compiled and on-chain bytecode do not match!
  warnings.warn(
Call trace for '0xe7d829cadfac1d0149b81606b121696ba4876f710356807b2ca6c92857990213':
Initial call cost  [21392 gas]
ProtocolFeePercentagesProvider.setFeeTypePercentage  0:1598  [8149 / 40687 gas]
│   
├── Vault.getAuthorizer  [STATICCALL]  228:324  [1849 gas]
│       ├── address: 0xBA12222222228d8Ba445958a75a0704d566BF2C8
│       └── input arguments: None
│       
├── Authorizer.canPerform  [STATICCALL]  455:607  [1065 / 2088 gas]
│   │   ├── address: 0xA331D84eC860Bf466b4CdCcFb4aC09a1B43F3aE6
│   │   ├── input arguments:
│   │   │   ├── actionId: 0xbeb10dd1f094c0751dc69e30b35c3f37cee8a9303f6e7380c83d31adcba39ea8
│   │   │   ├── account: 0x2f237e7643a3bF6Ef265dd6FCBcd26a7Cc38dbAa
│   │   │   └── : 0x1802953277FD955f9a254B80Aa0582f193cF1d77
│   │   └── return value: True
│   │   
│   └── AccessControl.hasRole  533:581  [118 / 1023 gas]
│       └── EnumerableSet.contains  550:572  [905 gas]
└── <UnknownContract>.0x38e9922e  [CALL]  841:1553  [24664 / 28601 gas]
    │   ├── address: 0xce88686553686DA562CE7Cea497CE749DA109f9F
    │   ├── value: 0
    │   └── calldata: 0x38e9922e00000000000000000000000000000000000000000000000006f05b59d3b20000
    │   
    │   
    ├── Vault.getAuthorizer  [STATICCALL]  1018:1114  [1849 gas]
    │       ├── address: 0xBA12222222228d8Ba445958a75a0704d566BF2C8
    │       └── input arguments: None
    │       
    └── Authorizer.canPerform  [STATICCALL]  1265:1417  [1065 / 2088 gas]
        │   ├── address: 0xA331D84eC860Bf466b4CdCcFb4aC09a1B43F3aE6
        │   ├── input arguments:
        │   │   ├── actionId: 0xb28b769768735d011b267f781c3be90bce51d5059ba015bc7a28b3e882fb2083
        │   │   ├── account: 0x1802953277FD955f9a254B80Aa0582f193cF1d77
        │   │   └── : 0xce88686553686DA562CE7Cea497CE749DA109f9F
        │   └── return value: True
        │   
        └── AccessControl.hasRole  1343:1391  [118 / 1023 gas]
            └── EnumerableSet.contains  1360:1382  [905 gas]
```

File: BIPs/BIP-303/BIP-303A.json [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+        "payable": false
+      },
+      "contractInputsValues": {
+        "feeType": "0",
+        "newValue": "500000000000000000"
```
Comment (Tritium-VLK) [inline BIPs/BIP-303/BIP-303A.json:28]: >>> 500000000000000000/10**18
0.5

File: BIPs/BIP-303/BIP-303A.json [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+    {
+      "to": "0x1802953277FD955f9a254B80Aa0582f193cF1d77",
+      "value": "0",
+      "data": null,
+      "contractMethod": {
+        "inputs": [
+          { "internalType": "uint256", "name": "feeType", "type": "uint256" },
+          { "internalType": "uint256", "name": "newValue", "type": "uint256" }
+        ],
+        "name": "setFeeTypePercentage",
```
Comment (Tritium-VLK) [inline BIPs/BIP-303/BIP-303A.json:40]:
```
>>> tx = ProtocolFeesPercentagesProvider.setFeeTypePercentage(2, 500000000000000000, {'from': daomsig})
tx.call_trace(expand=True)
 
Transaction sent: 0xe64df6bb0fa17de81e26a6c88c30428b3bc1f4fdcf60babb1a8462bd30ad9b06
  Gas price: 0.0 gwei   Gas limit: 20000000   Nonce: 10
  ProtocolFeePercentagesProvider.setFeeTypePercentage confirmed   Block: 387621   Gas used: 39547 (0.20%)

Call trace for '0xe64df6bb0fa17de81e26a6c88c30428b3bc1f4fdcf60babb1a8462bd30ad9b06':
Initial call cost  [21404 gas]
ProtocolFeePercentagesProvider.setFeeTypePercentage  0:911  [14206 / 18143 gas]
│   
├── Vault.getAuthorizer  [STATICCALL]  228:324  [1849 gas]
│       ├── address: 0xBA12222222228d8Ba445958a75a0704d566BF2C8
│       └── input arguments: None
│       
└── Authorizer.canPerform  [STATICCALL]  455:607  [1065 / 2088 gas]
    │   ├── address: 0xA331D84eC860Bf466b4CdCcFb4aC09a1B43F3aE6
    │   ├── input arguments:
    │   │   ├── actionId: 0xbeb10dd1f094c0751dc69e30b35c3f37cee8a9303f6e7380c83d31adcba39ea8
    │   │   ├── account: 0x2f237e7643a3bF6Ef265dd6FCBcd26a7Cc38dbAa
    │   │   └── : 0x1802953277FD955f9a254B80Aa0582f193cF1d77
    │   └── return value: True
    │   
    └── AccessControl.hasRole  533:581  [118 / 1023 gas]
        └── EnumerableSet.contains  550:572  [905 gas]
```

File: BIPs/BIP-303/BIP-303B.json [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+    {
+      "to": "0xA331D84eC860Bf466b4CdCcFb4aC09a1B43F3aE6",
+      "value": "0",
+      "data": null,
+      "contractMethod": {
+        "inputs": [
+          { "internalType": "bytes32", "name": "role", "type": "bytes32" },
+          { "internalType": "address", "name": "account", "type": "address" }
+        ],
+        "name": "grantRole",
```
Comment (Tritium-VLK) [inline BIPs/BIP-303/BIP-303B.json:23]:
```
 roleList = ["0x28eae42992658aefe28d2ca017a285a62376e3196f2ea3207d6e25d2dd31822d"]
tx = Authorizer.grantRoles(roleList, daomsig, {'from': daomsig})
tx.call_trace(expand=True)
Transaction sent: 0x304c3c79c4585decb0fc50b25b91087ab211c3c2fd5b941239babc1d56f930ca
  Gas price: 0.0 gwei   Gas limit: 20000000   Nonce: 11
  Authorizer.grantRoles confirmed   Block: 387622   Gas used: 90604 (0.45%)

Call trace for '0x304c3c79c4585decb0fc50b25b91087ab211c3c2fd5b941239babc1d56f930ca':
Initial call cost  [22224 gas]
Authorizer.grantRoles  0:436  [757 / 68380 gas]
└── AccessControl.grantRole  194:416  [953 / 67623 gas]
    ├── AccessControl.hasRole  216:264  [118 / 1023 gas]
    │   └── EnumerableSet.contains  233:255  [905 gas]
    ├── _require  268:276  [30 gas]
    └── AccessControl._grantRole  282:412  [2038 / 65617 gas]
        └── EnumerableSet.add  299:390  [62674 / 63579 gas]
            └── EnumerableSet.contains  306:328  [905 gas]
```

File: BIPs/BIP-303/BIP-303B.json [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+            "internalType": "contract ILiquidityGaugeFactory",
+            "name": "factory",
+            "type": "address"
+          }
+        ],
+        "name": "addGaugeFactory",
+        "payable": false
+      },
+      "contractInputsValues": {
+        "factory": "0x2498A2B0d6462d2260EAC50aE1C3e03F4829BA95"
```
Comment (Tritium-VLK) [inline BIPs/BIP-303/BIP-303B.json:47]:
```
>>> a.reversebook["0x2498A2B0d6462d2260EAC50aE1C3e03F4829BA95"]
'20230316-child-chain-gauge-factory-v2/ChildChainGaugeFactory'
```

File: BIPs/BIP-303/BIP-303B.json [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+      "data": null,
+      "contractMethod": {
+        "inputs": [
+          {
+            "internalType": "contract ILiquidityGaugeFactory",
+            "name": "factory",
+            "type": "address"
+          }
+        ],
+        "name": "addGaugeFactory",
```
Comment (Tritium-VLK) [inline BIPs/BIP-303/BIP-303B.json:43]:
``` tx = L2BalancerPseudoMinter.addGaugeFactory("0x2498A2B0d6462d2260EAC50aE1C3e03F4829BA95", {'from': daomsig})
Transaction sent: 0xfbe181d7e27693e929b0fd773f65604702c3bb95cebed347fb41f1bb4d9d44d2
  Gas price: 0.0 gwei   Gas limit: 20000000   Nonce: 12
  L2BalancerPseudoMinter.addGaugeFactory confirmed   Block: 387623   Gas used: 50856 (0.25%)

>>> tx.call_trace(expand=True)
 
Call trace for '0xfbe181d7e27693e929b0fd773f65604702c3bb95cebed347fb41f1bb4d9d44d2':
Initial call cost  [21432 gas]
L2BalancerPseudoMinter.addGaugeFactory  0:634  [25487 / 29424 gas]
│   
├── Vault.getAuthorizer  [STATICCALL]  183:279  [1849 gas]
│       ├── address: 0xBA12222222228d8Ba445958a75a0704d566BF2C8
│       └── input arguments: None
│       
└── Authorizer.canPerform  [STATICCALL]  379:531  [1065 / 2088 gas]
    │   ├── address: 0xA331D84eC860Bf466b4CdCcFb4aC09a1B43F3aE6
    │   ├── input arguments:
    │   │   ├── actionId: 0x28eae42992658aefe28d2ca017a285a62376e3196f2ea3207d6e25d2dd31822d
    │   │   ├── account: 0x2f237e7643a3bF6Ef265dd6FCBcd26a7Cc38dbAa
    │   │   └── : 0x475D18169BE8a89357A9ee3Ab00ca386d20fA229
    │   └── return value: True
    │   
    └── AccessControl.hasRole  457:505  [118 / 1023 gas]
        └── EnumerableSet.contains  474:496  [905 gas]
>>>


```

File: BIPs/BIP-303/BIP-303B.json [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+    {
+      "to": "0xA331D84eC860Bf466b4CdCcFb4aC09a1B43F3aE6",
+      "value": "0",
+      "data": null,
+      "contractMethod": {
+        "inputs": [
+          { "internalType": "bytes32", "name": "role", "type": "bytes32" },
+          { "internalType": "address", "name": "account", "type": "address" }
+        ],
+        "name": "renounceRole",
```
Comment (Tritium-VLK) [inline BIPs/BIP-303/BIP-303B.json:59]:
```
>>> tx = Authorizer.revokeRoles(roleList, daomsig, {'from': daomsig})
tx.call_trace(expand=True)
Transaction sent: 0xc6c6dc42bd946a1b96f3ddd6fca45e41ed71b83df4d9314caa76c668ce15ae08
  Gas price: 0.0 gwei   Gas limit: 20000000   Nonce: 13
  Authorizer.revokeRoles confirmed   Block: 387624   Gas used: 24955 (0.12%)

Call trace for '0xc6c6dc42bd946a1b96f3ddd6fca45e41ed71b83df4d9314caa76c668ce15ae08':
Initial call cost  [-2730 gas]
Authorizer.revokeRoles  0:514  [692 / -17315 gas]
└── AccessControl.revokeRole  180:494  [953 / -18007 gas]
    ├── AccessControl.hasRole  202:250  [118 / 1023 gas]
    │   └── EnumerableSet.contains  219:241  [905 gas]
    ├── _require  254:262  [30 gas]
    └── AccessControl._revokeRole  268:490  [2038 / -20013 gas]
        └── EnumerableSet.remove  285:468  [-22051 gas]
```

File: BIPs/BIP-303/BIP-303C.json [hunk 14]
```diff
@@ -0,0 +1,68 @@
+{
```
Comment (Tritium-VLK) [inline BIPs/BIP-303/BIP-303C.json:1]:
```

>>> roleList = ["0x1cbb503dcc0f4acaedf71a098211ff8b15a220fc26a6974a8d9deaab040fa6e0"]
tx = Authorizer.grantRoles(roleList, daomsig, {'from': daomsig})
tx.call_trace(expand=True)
vault = Contract("0xBA12222222228d8Ba445958a75a0704d566BF2C8")
tx = vault.setAuthorizer("0x8df317a729fcaA260306d7de28888932cb579b88", {'from': daomsig})
tx.call_trace(expand=True)
tx = Authorizer.revokeRoles(roleList, daomsig, {'from': daomsig})
tx.call_trace(expand=True)
Transaction sent: 0x7137e783a87bc47dfcdf1ba1a7a4285d2b388eb17f6cdefee6a40a75bac9e425
  Gas price: 0.0 gwei   Gas limit: 20000000   Nonce: 16
  Authorizer.grantRoles confirmed   Block: 387627   Gas used: 45604 (0.23%)

Call trace for '0x7137e783a87bc47dfcdf1ba1a7a4285d2b388eb17f6cdefee6a40a75bac9e425':
Initial call cost  [22224 gas]
Authorizer.grantRoles  0:436  [757 / 23380 gas]
└── AccessControl.grantRole  194:416  [953 / 22623 gas]
    ├── AccessControl.hasRole  216:264  [118 / 1023 gas]
    │   └── EnumerableSet.contains  233:255  [905 gas]
    ├── _require  268:276  [30 gas]
    └── AccessControl._grantRole  282:412  [2038 / 20617 gas]
        └── EnumerableSet.add  299:390  [17674 / 18579 gas]
            └── EnumerableSet.contains  306:328  [905 gas]
Transaction sent: 0x9eb56b2563c169b7f13093721a251fe1928dd2792c5071a862174f5262e0803d
  Gas price: 0.0 gwei   Gas limit: 20000000   Nonce: 17
  Vault.setAuthorizer confirmed   Block: 387628   Gas used: 35821 (0.18%)

Call trace for '0x9eb56b2563c169b7f13093721a251fe1928dd2792c5071a862174f5262e0803d':
Initial call cost  [17232 gas]
Vault.setAuthorizer  0:589  [16501 / 18589 gas]
│   
└── Authorizer.canPerform  [STATICCALL]  284:436  [1065 / 2088 gas]
    │   ├── address: 0xA331D84eC860Bf466b4CdCcFb4aC09a1B43F3aE6
    │   ├── input arguments:
    │   │   ├── actionId: 0x1cbb503dcc0f4acaedf71a098211ff8b15a220fc26a6974a8d9deaab040fa6e0
    │   │   ├── account: 0x2f237e7643a3bF6Ef265dd6FCBcd26a7Cc38dbAa
    │   │   └── : 0xBA12222222228d8Ba445958a75a0704d566BF2C8
    │   └── return value: True
    │   
    └── AccessControl.hasRole  362:410  [118 / 1023 gas]
        └── EnumerableSet.contains  379:401  [905 gas]
Transaction sent: 0x2d1ef16e9088b76556e7f8d8a9b063349e94c4f4e9c81941b54bda9faeab78d4
  Gas price: 0.0 gwei   Gas limit: 20000000   Nonce: 18
  Authorizer.revokeRoles confirmed   Block: 387629   Gas used: 24955 (0.12%)

Call trace for '0x2d1ef16e9088b76556e7f8d8a9b063349e94c4f4e9c81941b54bda9faeab78d4':
Initial call cost  [-2730 gas]
Authorizer.revokeRoles  0:514  [692 / -17315 gas]
└── AccessControl.revokeRole  180:494  [953 / -18007 gas]
    ├── AccessControl.hasRole  202:250  [118 / 1023 gas]
    │   └── EnumerableSet.contains  219:241  [905 gas]
    ├── _require  254:262  [30 gas]
    └── AccessControl._revokeRole  268:490  [2038 / -20013 gas]
        └── EnumerableSet.remove  285:468  [-22051 gas]
```

**Review by Tritium-VLK — CHANGES_REQUESTED** [review]
Need to add 1 msig address.

**Review by Tritium-VLK — CHANGES_REQUESTED** [review]
Need to add 1 msig address.

**Review by Tritium-VLK — APPROVED** [review]

**Review by Tritium-VLK — APPROVED** [review]

**Pull request merged**